pkg/controller/dns/recordset/managed.go

```diff
@@ -0,0 +1,181 @@
+	creds, auth, err := azureclients.GetAuthInfo(ctx, c.client, mg)
+	if err != nil {
+		return nil, err
+	}
+	cl := dns.NewRecordSetsClient(creds[azureclients.CredentialsKeySubscriptionID])
+	cl.Authorizer = auth
+	return &external{
+		client: dnsclients.NewRecordSetClient(cl),
+	}, nil
+}
+
+type external struct {
+	client dnsclients.RecordSetAPI
+}
+
+func (e *external) Observe(ctx context.Context, mg resource.Managed) (managed.ExternalObservation, error) {
+	r, ok := mg.(*v1alpha1.RecordSet)
+	if !ok {
+		return managed.ExternalObservation{}, errors.New(errNotDNSRecordSet)
+	}
+
+	az, err := e.client.Get(ctx, r)
+	if azureclients.IsNotFound(err) {
+		return managed.ExternalObservation{ResourceExists: false}, nil
+	}
+	if err != nil {
+		return managed.ExternalObservation{}, errors.Wrap(err, errGetDNSRecordSet)
+	}
+
+	r.Spec.ForProvider.Metadata = azureclients.LateInitializeStringMap(r.Spec.ForProvider.Metadata, az.Metadata)
+
+	dnsclients.UpdateRecordSetStatusFromAzure(r, az)
+
+	if r.Status.AtProvider.ProvisioningState != dnsclients.RecordSetSuccessfulState {
+		return managed.ExternalObservation{ResourceExists: true, ResourceUpToDate: true, ConnectionDetails: managed.ConnectionDetails{}}, nil
+	}
+
+	r.SetConditions(xpv1.Available())
+
+	o := managed.ExternalObservation{
+		ResourceExists:    true,
+		ResourceUpToDate:  dnsclients.RecordSetIsUpToDate(&r.Spec.ForProvider, az.RecordSetProperties),
+		ConnectionDetails: managed.ConnectionDetails{},
+	}
+
+	return o, nil
+}
+
+func (e *external) Create(ctx context.Context, mg resource.Managed) (managed.ExternalCreation, error) {
+	r, ok := mg.(*v1alpha1.RecordSet)
+	if !ok {
+		return managed.ExternalCreation{}, errors.New(errNotDNSRecordSet)
+	}
+
+	r.Status.SetConditions(xpv1.Creating())
+
+	if err := e.client.CreateOrUpdate(ctx, r); err != nil {
+		return managed.ExternalCreation{}, errors.Wrap(err, errCreateDNSRecordSet)
+	}
+
+	return managed.ExternalCreation{}, nil
+}
+
+func (e *external) Update(ctx context.Context, mg resource.Managed) (managed.ExternalUpdate, error) {
+	r, ok := mg.(*v1alpha1.RecordSet)
+	if !ok {
+		return managed.ExternalUpdate{}, errors.New(errNotDNSRecordSet)
+	}
+
+	az, err := e.client.Get(ctx, r)
+	if err != nil {
+		return managed.ExternalUpdate{}, errors.Wrap(err, errGetDNSRecordSet)
+	}
+
+	dnsclients.UpdateRecordSetStatusFromAzure(r, az)
+
+	if err := e.client.CreateOrUpdate(ctx, r); err != nil {
+		return managed.ExternalUpdate{}, errors.Wrap(err, errUpdateDNSRecordSet)
+	}
+
+	return managed.ExternalUpdate{}, nil
+}
+
+func (e *external) Delete(ctx context.Context, mg resource.Managed) error {
+	r, ok := mg.(*v1alpha1.RecordSet)
+	if !ok {
+		return errors.New(errNotDNSRecordSet)
+	}
+
+	az, err := e.client.Get(ctx, r)
+	if err != nil {
+		return errors.Wrap(err, errGetDNSRecordSet)
+	}
+
+	dnsclients.UpdateRecordSetStatusFromAzure(r, az)
+
+	mg.SetConditions(xpv1.Deleting())
```

Comment:
nit: Please check the managed reconciler. This can potentially be done there.

Reply:
I checked this. As you said that, managed reconciler does this operation. So I removed this line.